### PR TITLE
nextcloudPackages: update

### PR DESCRIPTION
**Pull request by dotlambda**
<details>
<summary>
Blocked by nextcloud/notify_push#384.

Currently `nixosTests.nextcloud.with-postgresql-and-redis28` fails.
</summary>

```
nextcloud # [   89.976758] systemd[1]: Starting Push daemon for Nextcloud clients...
nextcloud # [   90.073230] nextcloud-notify_push-start[1629]: Error:   × Failed to parse database configuration: error with configuration: empty
nextcloud # [   90.075955] nextcloud-notify_push-start[1629]:   │ host
nextcloud # [   90.077984] nextcloud-notify_push-start[1629]:   ├─▶ error with configuration: empty host
nextcloud # [   90.085820] nextcloud-notify_push-start[1629]:   ╰─▶ empty host
nextcloud # [   90.088663] systemd[1]: nextcloud-notify_push.service: Main process exited, code=exited, status=1/FAILURE
nextcloud # [   92.769143] nextcloud-notify_push-post-start[1630]: ✓ redis is configured
nextcloud # [   92.770434] nextcloud-notify_push-post-start[1630]: 🗴 using unencrypted http for push server is strongly discouraged
nextcloud # [   93.656747] nginx[939]: 2024/01/29 04:30:06 [crit] 939#939: *18 connect() to unix:/run/nextcloud-notify_push/sock failed (2: No such file or directory) while connecting to upstream, client: ::1, server: nextcloud, request: "GET /push/test/cookie HTTP/1.1", upstream: "http://unix:/run/nextcloud-notify_push/sock:/push/test/cookie", host: "nextcloud"
nextcloud # [   93.685838] nextcloud-notify_push-post-start[1630]: 🗴 can't connect to push server: Server error: `GET http://nextcloud/push/test/cookie` resulted in a `502 Bad Gateway` response:
nextcloud # [   93.693343] nextcloud-notify_push-post-start[1630]: <html>
nextcloud # [   93.694564] nextcloud-notify_push-post-start[1630]: <head><title>502 Bad Gateway</title></head>
nextcloud # [   93.702964] nextcloud-notify_push-post-start[1630]: <body>
nextcloud # [   93.708752] nextcloud-notify_push-post-start[1630]: <center><h1>502 Bad Gateway</h1></center>
nextcloud # [   93.711764] nextcloud-notify_push-post-start[1630]: <hr><center>ngin (truncated...)
nextcloud # [   93.777359] systemd[1]: nextcloud-notify_push.service: Control process exited, code=exited, status=1/FAILURE
nextcloud # [   93.779970] systemd[1]: nextcloud-notify_push.service: Failed with result 'exit-code'.
nextcloud # [   93.789056] systemd[1]: Failed to start Push daemon for Nextcloud clients.

```
</details>

## Description of changes
Update all packages and remove the unmaintained keeweb.
cc @onny who added keeweb in https://github.com/NixOS/nixpkgs/pull/197373

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

closes https://github.com/NixOS/nixpkgs/pull/283455

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by SuperSandro2000 — APPROVED** [review]
Diff otherwise LGTM, let's wait for a big fix release for notify push

**Review by teutat3s — APPROVED** [review]

File: pkgs/servers/nextcloud/notify_push.nix [hunk 1]
```diff
@@ -1,21 +1,22 @@
 { lib
 , fetchFromGitHub
+, fetchpatch2
```
Comment (onny) [inline pkgs/servers/nextcloud/notify_push.nix:3]: is this still required?

File: pkgs/servers/nextcloud/packages/nextcloud-apps.json [hunk 2]
```diff
@@ -12,7 +12,6 @@
 , "gpoddersync": "agpl3Only"
 , "groupfolders": "agpl3Plus"
 , "impersonate": "agpl3Plus"
-, "keeweb": "agpl3Plus"
```
Comment (onny) [inline pkgs/servers/nextcloud/packages/nextcloud-apps.json:15]: keeweb seems to get updates again https://github.com/jhass/nextcloud-keeweb/releases
there are also pushed to apps.nextcloud.com

Comment (teutat3s) [reply pkgs/servers/nextcloud/packages/nextcloud-apps.json:15]: The nextcloud app did get some updates, but the project itself did not: https://github.com/keeweb/keeweb/issues/2022

Comment (onny) [reply pkgs/servers/nextcloud/packages/nextcloud-apps.json:15]: ah i see okay

**Pull request merged**